# Pre-planned waypoint routes for drone swarm
# Each drone has a sequence of waypoints with position and timing information

waypoint_configs:
  rs1_drone_1:
    mission_name: "perimeter_patrol_south_west_brown"
    waypoints:
      - position: {x: -25.2, y: 6.7, z: 19.0}
        dwell_time: 0.2
      - position: {x: -25.5, y: -6.6, z: 19.0}
        dwell_time: 0.2
      - position: {x: -12.3, y: -6.6, z: 19.0}
        dwell_time: 0.2
      - position: {x: -3.0, y: 7.8, z: 19.0}
        dwell_time: 0.2
      - position: {x: -18.0, y: 12.4, z: 19.0}
        dwell_time: 0.2

  rs1_drone_2:
    mission_name: "perimeter_patrol_north_west_light_green"
    waypoints:
      - position: {x: -18.99, y: 14.61, z: 19.0}
        dwell_time: 2.0
      - position: {x: 6.3, y: 3.0, z: 19.0}
        dwell_time: 2.0
      - position: {x: 9.4, y: 13.9, z: 19.0}
        dwell_time: 2.0
      - position: {x: 16.9, y: 25.9, z: 19.0}
        dwell_time: 2.0
      - position: {x: 20.5, y: 11.7, z: 19.0}
        dwell_time: 2.0
      - position: {x: 23.9, y: 4.4, z: 19.0}
        dwell_time: 2.0
      - position: {x: 24.3, y: -8.9, z: 19.0}
        dwell_time: 2.0
      - position: {x: 12.5, y: -5.5, z: 19.0}
        dwell_time: 2.0

  rs1_drone_3:
    mission_name: "perimeter_patrol_north_east_green"
    waypoints:
      - position: {x: -19.0, y: 14.61, z: 16.0}
        dwell_time: 2.0
      - position: {x: 2.1, y: -7.0, z: 5.0}
        dwell_time: 2.0
      - position: {x: 0.6, y: -26.4, z: 5.0}
        dwell_time: 2.0
      - position: {x: 6.0, y: -29.2, z: 5.0}
        dwell_time: 2.0
      - position: {x: 10.8, y: -22.7, z: 5.0}
        dwell_time: 2.0
      - position: {x: 18.6, y: -27.9, z: 5.0}
        dwell_time: 2.0
      - position: {x: 25.1, y: -20.0, z: 5.0}
        dwell_time: 2.0
      - position: {x: -19.9, y: -18.6, z: 5.0}
        dwell_time: 2.0
      - position: {x: 13.8, y: -20.1, z: 5.0}
        dwell_time: 2.0
      - position: {x: 8.2, y: -16.8, z: 5.0}
        dwell_time: 2.0

  rs1_drone_4:
    mission_name: "diagonal_sweep"
    waypoints:
      - position: {x: -18.99, y: 14.61, z: 15.63}
        dwell_time: 2.0
      - position: {x: 5.0, y: -5.0, z: 10.0}
        dwell_time: 2.0
      - position: {x: 10.0, y: -8.0, z: 10.0}
        dwell_time: 1.5
      - position: {x: 15.0, y: -10.0, z: 10.0}
        dwell_time: 3.0
      - position: {x: 8.0, y: -15.0, z: 10.0}
        dwell_time: 2.5
      - position: {x: 5.0, y: -5.0, z: 10.0}
        dwell_time: 3.0

  rs1_drone_5:
    mission_name: "figure_eight_pattern"
    waypoints:
      - position: {x: 6.0, y: -6.0, z: 10.0}
        dwell_time: 1.0
      - position: {x: 12.0, y: -3.0, z: 10.0}
        dwell_time: 1.0
      - position: {x: 15.0, y: -6.0, z: 10.0}
        dwell_time: 2.0
      - position: {x: 12.0, y: -9.0, z: 8.0}
        dwell_time: 1.0
      - position: {x: 6.0, y: -6.0, z: 8.0}
        dwell_time: 1.0
      - position: {x: 9.0, y: -12.0, z: 8.0}
        dwell_time: 2.0
      - position: {x: 6.0, y: -6.0, z: 10.0}
        dwell_time: 3.0

mission_params:
  takeoff_altitude: 2.0
  landing_speed: 0.5
  waypoint_tolerance: 0.8
  max_velocity: 3.0
  loop_missions: true
  helipad_location:
    x: 0.0
    y: 0.0
    z: 0.0
  # fire_bag_location: {0, 0}